lib/components/Announcement/Announcement.docs.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, Fragment } from 'react';
+import { ComponentDocs } from '../../../site/src/types';
+import { Announcement } from './Announcement';
+import { Text } from '../Text/Text';
+import { Button } from '../Button/Button';
+import { Box } from '../Box/Box';
+import { TextLink } from '../TextLink/TextLink';
+import { Strong } from '../Strong/Strong';
+import { Stack } from '../Stack/Stack';
+import { Heading } from '../Heading/Heading';
+import { InlineCode } from '../../../site/src/App/InlineCode/InlineCode';
+import Code from '../../../site/src/App/Code/Code';
+
+const messages = [
+  'How did the web developer stir the soup?',
+  'With an aria ladle!',
+];
+
+const docs: ComponentDocs = {
+  category: 'Content',
+  description: (
+    <Stack space="large">
+      <Text>
+        Announces extra information to screen reader users without stealing
+        focus (using{' '}
+        <TextLink href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions">
+          aria-live
+        </TextLink>
+        ). This is useful for giving extra context to complex interactive
+        components. For example,{' '}
+        <TextLink href="/components/Autosuggest">Autosuggest</TextLink> uses
+        this to announce how many suggestions are available.
+      </Text>
+      <Text>
+        <Strong>Note:</Strong> Announcement should only be used for adding
+        context to interactive components.{' '}
+        <TextLink href="/components/HiddenVisually">HiddenVisually</TextLink>{' '}
+        should be used for adding context to static markup. If you&apos;re
+        unsure which to use contact{' '}
+        <TextLink href="https://seekchat.slack.com/channels/braid-support">
+          #braid-support
+        </TextLink>
+        .
+      </Text>
+      <Heading level="3">Usage</Heading>
+      <Text>
+        Announcement must be rendered <Strong>before</Strong> sending it any
```

Comment:
Given this restriction, did you think about making it a hook?

Reply:
I did. I actually originally implemented it as a hook and then moved to a component. Happy to discuss further but I feel the component is slightly more elegant. I think it fits more naturally as something you render rather than having to worry about effects and cleanups.